tests/pool_test.go

```diff
@@ -610,6 +611,68 @@ var _ = Describe("[sig-compute]VirtualMachinePool", decorators.SigCompute, func(
 		Entry("do not append index if set to false", pointer.P(false)),
 		Entry("append index if set to true", pointer.P(true)),
 	)
+
+	It("should respect maxUnavailable strategy during updates", func() {
+		newPool := newVirtualMachinePool()
+
+		replicas := 4
+		maxUnavailable := 1
+
+		doScale(newPool.Name, int32(replicas))
+		waitForVMIs(newPool.Namespace, newPool.Spec.Selector, replicas)
+
+		By(fmt.Sprintf("Setting maxUnavailable to %d", maxUnavailable))
+		patchData, err := patch.New(patch.WithReplace("/spec/maxUnavailable", intstr.FromInt(maxUnavailable))).GeneratePayload()
+		Expect(err).ToNot(HaveOccurred())
+		_, err = virtClient.VirtualMachinePool(newPool.ObjectMeta.Namespace).Patch(context.Background(), newPool.Name, types.JSONPatchType, patchData, metav1.PatchOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Making a VMI template change")
+		patchData, err = patch.New(patch.WithAdd(
+			fmt.Sprintf("/spec/virtualMachineTemplate/spec/template/metadata/labels/%s", newLabelKey), newLabelValue),
+		).GeneratePayload()
+		Expect(err).ToNot(HaveOccurred())
+		_, err = virtClient.VirtualMachinePool(newPool.ObjectMeta.Namespace).Patch(context.Background(), newPool.Name, types.JSONPatchType, patchData, metav1.PatchOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Verifying maxUnavailable constraint throughout the update process")
+		Consistently(func() error {
```

Comment:
I'm a little concerned this test will flake if the test cluster is too slow. Let's give it a go to see how good it works.